x-pack/solutions/security/plugins/security_solution/public/sourcerer/containers/index.tsx

```diff
@@ -64,14 +66,20 @@ export const useSourcererDataView = (
   );
 
   useEffect(() => {
-    if (selectedDataView == null || missingPatterns.length > 0) {
+    if (!newDataViewPickerEnabled && (selectedDataView == null || missingPatterns.length > 0)) {
```

Comment:
Wouldn't we want to do an early return here, like so?
```typescript
if (!newDataViewPickerEnabled) {
  return;
}
```

I don't think we should ever need to check the `else if` condition no?

Reply:
yup! applied, thanks